edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Diagnostics/MetricFilter.cs

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+namespace Microsoft.Azure.Devices.Edge.Agent.Diagnostics
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Text;
+    using Newtonsoft.Json;
+
+    /// <summary>
+    /// This class acts as a pass through for a group of metrics.
+    /// It will exclude metrics based on a tag whitelist and blacklist.
+    /// After filtering, it will add or remove tags.
+    /// </summary>
+    public class MetricFilter
+    {
+        readonly Dictionary<string, string> tagsWhitelist;
+        readonly Dictionary<string, string> tagsBlacklist;
+        readonly Dictionary<string, string> tagsToAdd;
+        readonly List<string> tagsToRemove;
+
+        public MetricFilter(Dictionary<string, string> tagsWhitelist = null, Dictionary<string, string> tagsBlacklist = null, Dictionary<string, string> tagsToAdd = null, List<string> tagsToRemove = null)
```

Comment:
typically if all of these are optional, then a builder pattern works pretty well.